agent/consul/leader_connect_ca.go

```diff
@@ -503,6 +504,7 @@ func (c *CAManager) initializeRootCA(provider ca.Provider, conf *structs.CAConfi
 		req := structs.CARequest{
 			Op:     structs.CAOpSetConfig,
 			Config: conf,
+			Cas:    conf.ModifyIndex,
```

Comment:
I set `cas` to ModifyIndex in here because we are reading modifying and writing the config in one place, I'm not sure about this may be I need to set it to 0 to avoid the check.